crypto/comp/c_zlib.c

```diff
@@ -199,6 +203,66 @@ static ossl_ssize_t zlib_stateful_expand_block(COMP_CTX *ctx, unsigned char *out
     return (ossl_ssize_t)(olen - state->istream.avail_out);
 }
 
+/* ONESHOT COMPRESSION/DECOMPRESSION */
+
+static int zlib_oneshot_init(COMP_CTX *ctx)
+{
+    return 1;
+}
+
+static void zlib_oneshot_finish(COMP_CTX *ctx)
+{
+}
+
+static ossl_ssize_t zlib_oneshot_compress_block(COMP_CTX *ctx, unsigned char *out,
+                                                size_t olen, unsigned char *in,
+                                                size_t ilen)
+{
+    uLongf out_size = (uLongf)olen;
+    ossl_ssize_t ret = -1;
+
+    if (ilen == 0)
+        return 0;
+    if (compress(out, &out_size, in, ilen) != Z_OK)
+        return -1;
+
+    if (out_size > OSSL_SSIZE_MAX)
```

Comment:
Does this imply a memory trash if it got here?

Reply:
This check is here to make sure that the cast on the following line is safe.